### PR TITLE
#161967105 bookmark an article

### PR DESCRIPTION
**Pull request by denomash**
#### What does this PR do?

- Users should be able to bookmark an article.

#### Description of Task to be completed?

-  a user can bookmark an article.
-  a user can unbookmark an article.

#### How should this be manually tested?

- Clone this repo from the terminal

```
git clone https://github.com/andela/ah-code-titans-frontend.git
```

- cd into the projects folder
```
cd ah-code-titans-frontend
```

- Then Run `git fetch ft-bookmark-article-161967105` 

- checkout the branch: `git checkout ft-bookmark-article-161967105`

- Install all dependencies by running: `npm install` or `yarn`

- Run the app by running: `npm start`

#### What are the relevant pivotal tracker stories?
[#161967105](https://www.pivotaltracker.com/story/show/161967105)

**Checklist**

- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my own code

- [x] My changes generate no new warnings

####Screenshot
**UNBOOKMARKED ARTICLE**
screenshot 2019-01-30 at 15 04 40

**BOOKMARKED ARTICLE**
screenshot 2019-01-30 at 15 04 32

### CODE REVIEW

**Review by bettblake08 — APPROVED** [review]
👍

**Review by davishooly — APPROVED** [review]

**Review by jillo-abdullahi — COMMENTED** [review]
👍

**Review by jillo-abdullahi — APPROVED** [review]
👍

**Review by Bryan-Cee — APPROVED** [review]
👍

**Review by Darius-Ndubi — APPROVED** [review]
👍

File: src/api/axiosConfig.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import axios from "axios";
 
 const instance = axios.create({
-  baseURL: "https://ah-codetitans-staging.herokuapp.com/",
+  baseURL: process.env.REACT_APP_API,
```
Comment (justmesam) [inline src/api/axiosConfig.js]: wasn't this changed in a prior PR?

File: src/components/views/articles/SingleArticle.js [hunk 2]
```diff
@@ -7,19 +9,24 @@ import { connect } from "react-redux";
...
+      currentArticle: {},
+      bookmarked: props.bookmarked
     };
 
     this.checkIfAuthenticated = this.checkIfAuthenticated.bind(this);
+    this.bookmarkArticle = this.bookmarkArticle.bind(this);
+    this.unBookmarkArticle = this.unBookmarkArticle.bind(this);
   }
 
   componentDidMount = () => {
```
Comment (justmesam) [inline src/components/views/articles/SingleArticle.js]: why do we have this as an arrow function?

**Review by justmesam — CHANGES_REQUESTED** [review]

File: src/api/axiosConfig.js (hunk 1 above)
Comment (denomash) [reply src/api/axiosConfig.js]: Yes, it was. This change happened after rebasing develop.

File: src/components/views/articles/SingleArticle.js (hunk 2 above)
Comment (denomash) [reply src/components/views/articles/SingleArticle.js]: This happened after rebasing too, but I'll fix it

Comment (denomash) [reply src/components/views/articles/SingleArticle.js]: Fixed it.

**Review by justmesam — APPROVED** [review]

**Pull request merged**